evalml/pipelines/classification_pipeline.py

```diff
@@ -54,6 +54,8 @@ def fit(self, X, y):
 
         Raises:
             ValueError: If the number of unique classes in y are not appropriate for the type of pipeline.
+            TypeError: If the dtype is boolean but pd.NA exists in the series.
```

Comment:
From what I recall, I think the linter has a problem with not including it in the docstring because an error is being raised in the function that isn't being mentioned, even though that error is being handled. I'll take another look at this though